src/rgw/driver/rados/rgw_lc_tier.cc

```diff
@@ -531,6 +530,12 @@ void RGWLCCloudStreamPut::init_send_attrs(const DoutPrefixProvider *dpp,
   for (auto& hi : rest_obj.attrs) {
     if (keep_attr(hi.first)) {
       attrs.insert(hi);
+    } else {
+      std::string s1 = boost::algorithm::to_lower_copy(hi.first);
+      const char* k = std::strstr(s1.c_str(), "x-amz");
```

Comment:
there's a [boost::algorithm::istarts_with()](https://www.boost.org/doc/libs/1_79_0/doc/html/boost/algorithm/istarts_with.html) that looks perfect for this

Reply:
I observed that user defined attrs are stored with extra 'user-' prefix in RGW (for eg., user-x-amz-meta-key1). So I  used std::strstr here to split and fetch only the original key (x-amz-meta-key1) to be added as header.